src/metadata/persistence/ontologies.clj

```diff
@@ -1,101 +1,122 @@
 
 (defn get-ontology-details
   [ontology-version]
-  (first (select :ontologies
-                 (fields :version
-                         :iri
-                         :created_by
-                         :created_on)
-                 (where {:version ontology-version}))))
+  (let [q (-> (ontologies-base-query ontology-columns)
+              (h/where [:= :version ontology-version]))]
+    (plan/select-one! ds ontology-columns (sql/format q))))
 
 (defn set-ontology-deleted
   [ontology-version deleted]
-  (sql/update :ontologies
-              (set-fields {:deleted deleted})
-              (where {:version ontology-version})))
+  (jsql/update! ds
+                (t "ontologies")
+                {:deleted deleted}
+                {:version ontology-version}))
 
 (defn list-ontologies
   []
-  (select :ontologies
-          (fields :version
-                  :iri
-                  :created_by
-                  :created_on)
-          (where {:deleted false})))
+  (let [q (-> (ontologies-base-query ontology-columns)
+              (h/where [:= :deleted false]))]
+    (plan/select! ds ontology-columns (sql/format q))))
+
+(def ontology-class-columns [:iri :label :description])
+(defn- ontology-classes-base-query
+  [cols]
+  (-> (apply h/select cols)
+      (h/from (t "ontology_classes"))))
 
 (defn add-classes
   [ontology-version classes]
   (let [class-values (map #(assoc % :ontology_version ontology-version) classes)]
     (when-not (empty? class-values)
-      (insert :ontology_classes (values class-values)))))
+      (let [new-values (mapv #(vector (:ontology_version %) (:iri %) (:label %) (:description %)) class-values)]
+        (jsql/insert-multi! ds
+                            (t "ontology_classes")
+                            [:ontology_version :iri :label :description]
+                            new-values)))))
 
 (defn get-classes
   [ontology-version]
-  (select :ontology_classes
-          (fields :iri
-                  :label
-                  :description)
-          (where {:ontology_version ontology-version})))
-
-(defn- search-classes-base
-  [ontology-version search-term]
-  (let [search-term (str "%" (format-query-wildcards search-term) "%")]
-    (-> (select* :ontology_classes)
-        (where {:ontology_version    ontology-version
-                (sqlfn lower :label) [like (sqlfn lower search-term)]}))))
+  (let [q (-> (ontology-classes-base-query ontology-class-columns)
+              (h/where [:= :ontology_version ontology-version]))]
+    (plan/select! ds ontology-class-columns (sql/format q))))
 
 (defn search-classes-subselect
   [ontology-version search-term]
-  (-> (search-classes-base ontology-version search-term)
-      (subselect (fields :iri))))
+  (let [search-term (str "%" (format-query-wildcards search-term) "%")]
+    (-> (h/select :iri)
+        (h/from (t "ontology_classes"))
+        (h/where [:= :ontology_version ontology-version]
+                 [:ilike :label search-term]))))
```

Comment:
We weren't using `search-classes-base` anywhere else, and I think it was an artifact of the `select*` vs. `select` vs `subselect` distinction in korma due to it handling both query generation and execution. With honeysql + next.jdbc they're already separate, so `h/select` handles all three.

Reply:
I also used `ilike` instead of lowercasing both sides, which is inconsistent with one other spot. I guess we should check if that affects indexing and maybe use the lowercasing if that's better, but dunno